lib/availability_status_listener.rb

```diff
@@ -59,6 +59,11 @@ def process_event(event)
     Rails.logger.error("Could not find #{model_class} with id #{record_id}")
   rescue ActiveRecord::RecordInvalid
     Rails.logger.error("Invalid status #{payload["status"]}")
+  rescue ActiveRecord::StatementInvalid, PG::UnableToSend
+    Rails.logger.error("Ran into error connecting to db - retrying.")
+    ActiveRecord::Base.establish_connection
+
+    retry
```

Comment:
I am just thinking whether we need count and set limit how many times we can retry but maybe we can try it without limit.